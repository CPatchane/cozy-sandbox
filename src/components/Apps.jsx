import React from 'react'

import { Button } from 'cozy-ui/react/Button'
import IntentOpener from 'cozy-ui/react/IntentOpener'

const Apps = () => (
  <div>
    <h1>Apps</h1>
    <IntentOpener
      action="INSTALL"
      doctype="io.cozy.apps"
      onComplete={res => alert('intent has completed ! ' + JSON.stringify(res))}
      onDismiss={() => alert('intent has been dismissed !')}
      options={{
        slug: 'trainline'
      }}
    >
      <Button>Install app Trainline</Button>
    </IntentOpener>
  </div>
)

export default Apps
