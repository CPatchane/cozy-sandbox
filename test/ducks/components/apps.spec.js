'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Apps from '../../../src/components/Apps'

Enzyme.configure({ adapter: new Adapter() })

describe('Apps', () => {
  it('should be rendered correctly', () => {
    const component = shallow(<Apps />).getElement()
    expect(component).toMatchSnapshot()
  })
})
