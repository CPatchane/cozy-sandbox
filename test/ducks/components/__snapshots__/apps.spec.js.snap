// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apps should be rendered correctly 1`] = `
<div>
  <h1>
    Apps
  </h1>
  <IntentOpener
    action="INSTALL"
    closable={true}
    doctype="io.cozy.apps"
    onComplete={[Function]}
    onDismiss={[Function]}
    options={
      Object {
        "slug": "trainline",
      }
    }
    tag="span"
  >
    <Button
      busy={false}
      className=""
      disabled={false}
      icon={null}
      label={null}
      onClick={null}
      theme=""
      type="submit"
    >
      Install app Trainline
    </Button>
  </IntentOpener>
</div>
`;
